app/gui2/src/workers/visualizationCompiler.ts

```diff
@@ -0,0 +1,174 @@
+import { parse, compileScript, compileStyle } from 'vue/compiler-sfc'
```

Comment:
This file really deserves a file-level doc explaining what the compiler is doing, preferably listing the transformation steps and explaining their purpose.